Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -304,3 +358,37 @@ private extension HTTPHeaders {
         return headers
     }
 }
+
+extension Array where Element == PackageController.DocumentationVersion {
+    subscript(reference reference: String) -> Element? {
+        first { "\($0.reference)" == reference }
+    }
+
+    func latestMajorVersions() -> Self {
+        let stableVersions = self.filter { version in
+            guard let semVer = version.reference.semVer else { return false }
+            return semVer.isStable
+        }
+        let groupedStableVersions = Dictionary.init(grouping: stableVersions) { version in
+            version.reference.semVer?.major
+        }
+
+        return groupedStableVersions.compactMap { key, versions -> Element? in
+            // If any of the references had a nil semVer then there could be a nil key in the dictionary.
+            guard let _ = key else { return nil }
+
+            // Filter down to only the largest semVer in each group.
+            let latestMajorStableVersion = versions
+                .compactMap { result -> (result: Element, semVer: SemanticVersion)? in
+                    guard let semVer = result.reference.semVer else { return nil }
+                    return (result: result, semVer: semVer)
+                }
+                .sorted(using: KeyPathComparator(\.semVer, order: .reverse))
```

Comment:
`KeyPathComparator` is not available on Linux it seems:

```
❌ /__w/SwiftPackageIndex-Server/SwiftPackageIndex-Server/Sources/App/Controllers/PackageController+routes.swift:392:24: cannot find 'KeyPathComparator' in scope
        .sorted(using: KeyPathComparator(\.reference.semVer))
```

Reply:
Ahh. I saw the tests failing last night with valid errors, but the tests passed on my local machine. I couldn't spend another minute in the heat of the office, though! I'll swap these out for regular sorted.

It's a shame, it's a really nice API.